rfcs/0000-occupied-capacity-increment-rate/0000-occupied-capacity-increment-rate.md

```diff
@@ -0,0 +1,120 @@
+---
+Number: 0000
+Category: Standards Track
+Status: Proposal
+Author: Jinyang Jiang
+Organization: Nervos Foundation
+Created: 2019-2-11
+---
+
+# Occupied capacity increment rate
+
+## Abstract
+
+Capacity is one of the core resources in CKB network, itself is just a number representing how much storage resource a user can use.
+
+Once a user decides to store something in CKB, the user must lock some capacity in exchange, the locked capacity which we called occupied capacity represents the actually used resource in the network.
+
+Occupied capacity affects CKB network in two ways:
+
+1. Single block occupied capacity: if a single block or transaction contains too much occupied capacity, it may affect the propagation of the block and may slow down the whole network.
+2. The whole network occupied capacity: that every full node must cost the same memory or disk to save these content, it incentives miners to be more centralized which does not make any sense for a decentralized network.
+
+So the solution is to limit occupied capacity in each block, to indirectly control the network's total occupied capacity and the increment rate of occupied capacity.
+
+This RFC proposes to make occupied capacity under control by adding two verification rules:
+
+1. limit the block size
+2. limit the increment of total network occupied capacity in block level
+
+
+## Specification
+
+Define three variables:
+
+* `MAX_BLOCK_BODY_BYTES`           - the max bytes of `commit_transactions` and `proposal_transactions` in a block, value: 10 MB
+* `MAX_BLOCK_UNCLES_BYTES`         - the max bytes of uncles blocks, value: 1 MB
+* `MAX_INCREMENT_OCCUPIED_CAPACITY` - the max increment occupied capacity of a block, value: 2 MB
+
+Add two verification rules to CKB consensus:
+
+### 1. Block size verification rule
+
+Unfortunately, we could not just put a block size limitation on CKB like Bitcoin.
+
+Because CKB has four fields defined in each block: `header`, `uncles`, `commit_transactions`, `proposal_transactions`, but CKB miner takes fees only from `commit_transactions`, they do not receive any incentives for submitting other parts. So a reasonable miner first packs `commit_transactions` into the block to make sure transactions fees as more as possible and immediately give up packs other parts once block size reaches the limitation.
+
+It is bad news because CKB consensus adjusts the network based on uncle blocks, CKB consensus will not properly work once miners give up to submit `uncles`.
+
+So instead of a simple block size limitation, we must separately limit the size of `uncles` from other parts to make CKB consensus work under the designed intention.
```

Comment:
Or incentivize uncles inclusion. This RFC should not be merged before we settled consensus details.

Reply:
I'll remove the "limit block size" part from the proposal and make another PR only include part of the "limit increase of occupied capacity".